tests/sqlparser_redshift.rs

```diff
@@ -196,3 +196,12 @@ fn test_create_view_with_no_schema_binding() {
     redshift_and_generic()
         .verified_stmt("CREATE VIEW myevent AS SELECT eventname FROM event WITH NO SCHEMA BINDING");
 }
+
+#[test]
+fn test_select_top() {
```

Comment:
can we move the test to common and run the scenarios via `all_dialects_where(|| expects_top_before-distinct()`? (same for distinct row)